pkg/proxy/ipvs/proxier.go

```diff
@@ -1270,11 +1271,12 @@ func (proxier *Proxier) writeIptablesRules() {
 		writeLine(proxier.natRules, append(dstLocalOnlyArgs, "-j", "ACCEPT")...)
 	}
 
-	// mark masq for KUBE-NODE-PORT
-	writeLine(proxier.natRules, []string{
-		"-A", string(KubeNodePortChain),
-		"-j", string(KubeMarkMasqChain),
-	}...)
+	// -A KUBE-SERVICES  -m addrtype  --dst-type LOCAL -j KUBE-NODE-PORT
+	args = append(args[:0],
+		"-A", string(kubeServicesChain),
+		"-m", "addrtype", "--dst-type", "LOCAL",
+	)
+	writeLine(proxier.natRules, append(args, "-j", string(KubeNodePortChain))...)
```

Comment:
I don't think this is right. All IP in `kube-ipvs0` interface can match `dst-type LOCAL`.

Reply:
Please see the changes of KUBE-NODE-PORT chain. Only packets matching KUBE-NODE-PORT-{TCP,UDP} and not matching KUBE-NODE-PORT-{TCP,UDP}-LOCAL are sent to KUBE-MARK-MASQ.

Reply:
What I'm trying to say is, for example, if we have a service with clusterIP address `172.17.0.1:443`, then iptables with treat `172.17.0.1` as `LOCAL`, traffic to `172.17.0.1:443` will match this `-A KUBE-SERVICES  -m addrtype  --dst-type LOCAL -j KUBE-NODE-PORT` rule, then if we have a service with nodeport `443`, error could happen.

Reply:
The potential  error exists before this PR too.
Why there's no problem is because the match for ClusterIP is prior to the match for NodePort. This condition is kept after this PR.

Reply:
clusterIP address 172.17.0.1:443 will be matched by the preceding ClusterIP rule:
```
-A KUBE-SERVICES -m comment --comment "Kubernetes service cluster ip + port for masquerade purpose" -m set --match-set KUBE-CLUSTER-IP dst,dst -j KUBE-MARK-MASQ
```,  not by the rule you mentioned.

Even someone disordered these two rules, say,  "-A KUBE-SERVICES -m addrtype --dst-type LOCAL -j KUBE-NODE-PORT" is matched first, nothing will happen in KUBE-NODE-PORT chain, and the processing will return from KUBE-NODE-PORT chain to KUBE-SERVICES chain, and hitting KUBE-CLUSTER-IP match.

Reply:
@honkiko I don't think `dst-type LOCAL` is right too. 
Imagine a scenario: there are two nodes in cluster, named node A and node B. If there is a container in node A and it want to access svc by `<node B ip>:<nodePort>`, it will match:

```
-A OUTPUT -m comment --comment "kubernetes service portals" -j KUBE-SERVICES
-A KUBE-SERVICES -m comment --comment "Kubernetes service cluster ip + port for masquerade purpose" -m set --match-set KUBE-CLUSTER-IP dst,dst -j KUBE-MARK-MASQ
-A KUBE-SERVICES -m set --match-set KUBE-CLUSTER-IP dst,dst -j ACCEPT
-A KUBE-SERVICES -m addrtype --dst-type LOCAL -j KUBE-NODE-PORT
```

If you set `-m addrtype --dst-type LOCAL`, it will not match `KUBE-NODE-PORT` and the packets will not masq in some cases.

Reply:
Sorry,it is my mistake. `all nodes can communicate with all containers (and vice-versa) without NAT` is Kubernetes model fundamental requirements. So above scenario will work ok.

Reply:
@honkiko Sorry for the delay replaying. In normal case, if we didn't specified masq-all, traffic from in-cluster to another clusterIP should not be masquerade, cause we need preserve the source address information. 
But if matched `-A KUBE-SERVICES -m addrtype --dst-type LOCAL -j KUBE-NODE-PORT`, jumped to KUBE-NODE_PORT, then the traffic might be masquerade. Right?